tango/executor.py

```diff
@@ -27,26 +45,39 @@ def __init__(
         self.workspace = workspace
         self.include_package = include_package
 
-    def execute_step(self, step: "Step", is_uncacheable_leaf_step: bool = False) -> None:
+    def execute_step(self, step: "Step") -> None:
         # Import included packages to find registered components.
         if self.include_package is not None:
             for package_name in self.include_package:
                 import_extra_module(package_name)
 
-        if step.cache_results:
-            step.ensure_result(self.workspace)
-        elif is_uncacheable_leaf_step:
-            step.result(self.workspace)
+        step.result(self.workspace)
```

Comment:
This should still call `ensure_result()` for cacheable steps. That's an optimization that can potentially save a lot of disk IO.